frontend/src/DeviceDetailsPage/__tests__/DeviceDetailsTabTable.test.jsx

```diff
@@ -0,0 +1,141 @@
+          <div>
+            {DeviceDetailsTabTable.getPanelContents(dummyTabInfo, dummyDevice)}
+          </div>
+        );
+        expect(wrapper.find(DeviceDetailsConciseTable)).toHaveLength(1);
+      });
+      it('Returns a DeviceDetailsActivityPanel component if passed "Activity Log"', () => {
+        const dummyTabInfo = "Activity Log";
+        const dummyDevice = new DeviceInfo(1, 1, 1, 1, 1, 1, ["Hello"]);
+
+        wrapper = Enzyme.shallow(
+          <div>
+            {DeviceDetailsTabTable.getPanelContents(dummyTabInfo, dummyDevice)}
+          </div>
+        );
+        expect(wrapper.find(DeviceDetailsActivityPanel)).toHaveLength(1);
+      });
+      it('Returns a DeviceDetailsNotesPanel component if passed "Notes"', () => {
+        const dummyTabInfo = "Notes";
+        const dummyDevice = new DeviceInfo(1, 1, 1, 1, 1, 1, ["Hello"]);
+
+        wrapper = Enzyme.shallow(
+          <div>
+            {DeviceDetailsTabTable.getPanelContents(dummyTabInfo, dummyDevice)}
+          </div>
+        );
+        expect(wrapper.find(DeviceDetailsNotesPanel)).toHaveLength(1);
+      });
+      it("Returns a div component if passed an invalid value", () => {
+        const dummyTabInfo = "Not a valid Value";
+        const dummyDevice = new DeviceInfo(1, 1, 1, 1, 1, 1, ["Hello"]);
+
+        wrapper = Enzyme.shallow(
+          <div>
+            {DeviceDetailsTabTable.getPanelContents(dummyTabInfo, dummyDevice)}
+          </div>
+        );
+        expect(wrapper.children).toHaveLength(1);
+        expect(wrapper.childAt(0).text()).toEqual("Whoops not a valid value");
+      });
+    });
+  });
+  describe("non-static functions", () => {
+    describe("handleValueChange", () => {
+      it("Should call setState", () => {
+        const tabs = ["a", "b"];
+        const device = null;
+
+        wrapper = Enzyme.shallow(
+          <DeviceDetailsTabTable tabs={tabs} device={device} />
+        );
+
+        const expected = {
+          value: 69
+        };
+
+        wrapper.instance().handleValueChange(null, 69);
+        expect(wrapper.state()).toEqual(expected);
+      });
+    });
+  });
+  describe("render()", () => {
+    describe("Renders the correct number of Tabs and Tab Panels", () => {
+      it("when passed an array of length 2", () => {
+        const tabs = ["a", "b"];
+        const device = null;
+
+        wrapper = Enzyme.shallow(
+          <DeviceDetailsTabTable tabs={tabs} device={device} />
+        );
+
+        expect(wrapper.find(Tab)).toHaveLength(2);
+        expect(wrapper.find(TabPanelH)).toHaveLength(2);
+      });
+      it("when passed an array of length 5", () => {
+        const tabs = ["a", "b", "a", "b", "a"];
+        const device = null;
+
+        wrapper = Enzyme.shallow(
+          <DeviceDetailsTabTable tabs={tabs} device={device} />
+        );
+
+        expect(wrapper.find(Tab)).toHaveLength(5);
+        expect(wrapper.find(TabPanelH)).toHaveLength(5);
+      });
+    });
+
```

Comment:
i assume this line is extra since it's the only blank line before an `it` i've noticed so far